src/main/java/com/blamejared/crafttweaker/impl/util/CTMath.java

```diff
@@ -0,0 +1,206 @@
+
+    @ZenCodeType.Field
+    public static final double E = Math.E;
+
+    @ZenCodeType.Method
+    public static double max(double d1, double d2)
+    {
+        return Math.max(d1, d2);
+    }
+
+    @ZenCodeType.Method
+    public static float max(float f1, float f2)
+    {
+        return Math.max(f1, f2);
+    }
+
+    @ZenCodeType.Method
+    public static long max(long l1, long l2)
+    {
+        return Math.max(l1, l2);
+    }
+
+    @ZenCodeType.Method
+    public static int max(int i1, int i2)
+    {
+        return Math.max(i1, i2);
+    }
+
+    @ZenCodeType.Method
+    public static double min(double d1, double d2)
+    {
+        return Math.min(d1, d2);
+    }
+
+    @ZenCodeType.Method
+    public static float min(float f1, float f2)
+    {
+        return Math.min(f1, f2);
+    }
+
+    @ZenCodeType.Method
+    public static long min(long l1, long l2)
+    {
+        return Math.min(l1, l2);
+    }
+
+    @ZenCodeType.Method
+    public static int min(int i1, int i2)
+    {
+        return Math.min(i1, i2);
+    }
+
+    @ZenCodeType.Method
+    public static long floor(double d1)
+    {
+        return new Double(Math.floor(d1)).longValue();
+    }
+
+    @ZenCodeType.Method
+    public static long ceil(double d1)
+    {
+        return new Double(Math.ceil(d1)).longValue();
+    }
+
+    @ZenCodeType.Method
+    public static double abs(double d1)
+    {
+        return Math.abs(d1);
+    }
+
+    @ZenCodeType.Method
+    public static float abs(float f1)
+    {
+        return Math.abs(f1);
+    }
+
+    @ZenCodeType.Method
+    public static long abs(long l1)
+    {
+        return Math.abs(l1);
+    }
+
+    @ZenCodeType.Method
+    public static int abs(int i1)
+    {
+        return Math.abs(i1);
+    }
+
+    @ZenCodeType.Method
+    public static double sin(double d1)
+    {
+        return Math.sin(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double cos(double d1)
+    {
+        return Math.cos(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double tan(double d1)
+    {
+        return Math.tan(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double asin(double d1)
+    {
+        return Math.asin(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double acos(double d1)
+    {
+        return Math.acos(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double atan(double d1)
+    {
+        return Math.atan(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double sinh(double d1)
+    {
+        return Math.sinh(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double cosh(double d1)
+    {
+        return Math.cosh(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double tanh(double d1)
+    {
+        return Math.tanh(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double sqrt(double d1)
+    {
+        return Math.sqrt(d1);
+    }
+
+    @ZenCodeType.Method
+    public static double random()
```

Comment:
Ehhhh iirc devs don't want random in the main mod because many people will misuse it (use it on both client and server, causing desync).

Reply:
So can I add an `@OnlyIn(SERVER)` ?

Reply:
No, firstly that would mean nothing for CraftTweaker and actually cause more issues.

Secondly the issue is that most modpack makers don't know how randomness or scripts work, so they would do

```
addRecipe(<output> * random())
```
And wonder why it isn't random each time it is crafted. 

Also if they figure out it can only be used in events/ functions, you then need to explain the different threads (world.isRemote), which is usually a problem for some of them to understand.

I'm just speaking from experience of what people have tried to do, and having to explain why what they are doing *doesn't* work.

At the end of the day, I don't care if another mod adds random, but I would prefer not to do it because I don't want to have to deal with the questions and "bug" reports that come from it.

Reply:
Oh, I know that. And The random generator is also be gotten via `world.rand`. But there don't seem to be `IWorld` in CrT 1.16?!

Reply:
1.14+ came at a bad time, so we haven't had a chance to reimplement all the wrapper classes also it doesn't look like many people are making 1.16 modpacks, so there hasn't been a huge demand for all the wrappers